commercetools/custom_fields_test.go

```diff
@@ -66,3 +74,178 @@ func TestCustomFieldEncodeValue(t *testing.T) {
+						assert.EqualValues(t, "2023-08-29", customFields.Fields["Date-field"], fmt.Sprintf("Date-field unexpected value for %s resource.", customFieldResourceType))
+						assert.EqualValues(t, "20:22:11.123", customFields.Fields["Time-field"], fmt.Sprintf("Time-field unexpected value for %s resource.", customFieldResourceType))
+						assert.EqualValues(t, "2023-08-29T20:22:11.123Z", customFields.Fields["DateTime-field"], fmt.Sprintf("DateTime-field unexpected value for %s resource.", customFieldResourceType))
+						assert.EqualValues(t, map[string]interface{}{"id": productType.ID, "typeId": "product-type"}, customFields.Fields["Reference-field"], fmt.Sprintf("Reference-field unexpected value for %s resource.", customFieldResourceType))
+						assert.EqualValues(t, []any{"ENUM-1", "ENUM-3"}, customFields.Fields["Set-field"], fmt.Sprintf("Set-field unexpected value' for %s resource.", customFieldResourceType))
+						return nil
+					},
+				),
+			},
+		}
+
+		// Remove Custom fields from the resource one by one
+		for index := range customFieldTypes {
+			var customFieldTypesReduced = []string{}
+			for i := range customFieldTypes {
+				if i == index {
+					continue
+				}
+				customFieldTypesReduced = append(customFieldTypesReduced, customFieldTypes[i])
+			}
+			fieldTypeValue := customFieldTypes[index]
+
+			customFieldsAccTestSteps = append(customFieldsAccTestSteps, resource.TestStep{
+				Config: getResourceConfig(customFieldResourceType, resourceShortName, resourceKey, customFieldTypesReduced),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					func(s *terraform.State) error {
+						customFields, err := getResourceCustomFields(s, customFieldResourceType, resourceFullName)
+						if err != nil {
+							return err
+						}
+						assert.Nil(t, customFields.Fields[fmt.Sprintf("%s-field", fieldTypeValue)], fmt.Sprintf("%s-field expected to be removed.", fieldTypeValue))
+						return nil
+					},
+				),
+			})
+		}
+
+		// Remove all Custom fields from the resource
+		customFieldsAccTestSteps = append(customFieldsAccTestSteps, resource.TestStep{
+			Config: getResourceConfig(customFieldResourceType, resourceShortName, resourceKey, []string{}),
+			Check: resource.ComposeAggregateTestCheckFunc(
+				func(s *terraform.State) error {
+					customFields, err := getResourceCustomFields(s, customFieldResourceType, resourceFullName)
+					if err != nil {
+						return err
+					}
+					assert.Nil(t, customFields, fmt.Sprintf("%v-field expected to be nil.", customFields))
+					return nil
+				},
+			),
+		})
+
+		resource.Test(t, resource.TestCase{
+			PreCheck:  func() { testAccPreCheck(t) },
+			Providers: testAccProviders,
+			Steps:     customFieldsAccTestSteps,
+		})
+	}
+}
+
+func getResourceConfig(resourceType, resourceName, resourceKey string, customFields []string) string {
+	// Load templates
+	template, err := template.ParseGlob("test_templates/custom_fields_test/*")
```

Comment:
We might also want to prevent naming collision with `text/template` by using a different name than `template` here. Something like `tpl` should be fine